disaggregated_memory/boards/riscv_sst_board.py

```diff
@@ -0,0 +1,244 @@
+        )
+        self.local_memory = local_memory
+        self.remote_memory = self._remoteMemory.remote_memory
+
+        if processor.get_isa() != ISA.RISCV:
+            raise Exception(
+                "The RISCVBoard requires a processor using the"
+                "RISCV ISA. Current processor ISA: "
+                f"'{processor.get_isa().name}'."
+            )
+
+    @overrides(AbstractSystemBoard)
+    def get_memory(self) -> "AbstractMemory":
+        """Get the memory (RAM) connected to the board.
+
+        :returns: The memory system.
+        """
+        raise NotImplementedError
+
+    def get_local_memory(self) -> "AbstractMemory":
+        """Get the memory (RAM) connected to the board.
+        :returns: The local memory system.
+        """
+        return self._localMemory
+
+    def get_remote_memory(self) -> "AbstractMemory":
+        """Get the memory (RAM) connected to the board.
+        :returns: The remote memory system.
+        """
+        # raise Exception("cannot call this method")
+        return self._remoteMemory
+
+    @overrides(AbstractSystemBoard)
+    def get_mem_ports(self) -> Sequence[Tuple[AddrRange, Port]]:
+        return self.get_local_memory().get_mem_ports()
+
+    def get_remote_mem_ports(self) -> Sequence[Tuple[AddrRange, Port]]:
+        return [
+            (
+                self.get_remote_memory().physical_address_ranges,
+                self.get_remote_memory().port,
+            )
+        ]
+
+    @overrides(RiscvAbstractDMBoard)
+    def _incorporate_memory_range(self):
+        self.get_local_memory().set_memory_range(self._local_mem_ranges)
+
+    @overrides(RiscvAbstractDMBoard)
+    def get_default_kernel_args(self) -> List[str]:
+        return ["console=ttyS0", "root={root_value}", "init=/bin/bash", "rw"]
+
+    @overrides(RiscvAbstractDMBoard)
+    def _connect_things(self) -> None:
+        """Connects all the components to the board.
+
+        The order of this board is always:
+
+        1. Connect the memory.
+        2. Connect the cache hierarchy.
+        3. Connect the processor.
+
+        Developers may build upon this assumption when creating components.
+
+        Notes
+        -----
+
+        * The processor is incorporated after the cache hierarchy due to a bug
+        noted here: https://gem5.atlassian.net/browse/GEM5-1113. Until this
+        bug is fixed, this ordering must be maintained.
+        * Once this function is called `_connect_things_called` *must* be set
+        to `True`.
+        """
+
+        if self._connect_things_called:
+            raise Exception(
+                "The `_connect_things` function has already been called."
+            )
+
+        # Incorporate the memory into the motherboard.
+        self.get_local_memory().incorporate_memory(self)
+        # we need to find whether there is any external latency. if yes, then
+        # add xbar to add this latency.
+
+        if self.get_remote_memory().is_xbar_required():
```

Comment:
This xbar should bein the remote memory object and simply exposed as a single port. 

Let's generalize this by assuming the remote memory will always be a single port (i.e., never channeled). Then, you can make this function be exactly the same for sst/gem5 and arm/riscv.

In fact, there should be a "mixin" class that allows disaggregated memory.